script/script.js

```diff
@@ -0,0 +1,177 @@
+function displaySignUp() {
+	document.getElementById("signup").style.display="block";
+}
+
+function displayLogIn() {
+	document.getElementById("login").style.display="block";
+}
+
+function closeDialog() {
+	document.getElementById("signup").style.display="none";
+	document.getElementById("login").style.display="none";
+}
+
+var validateUsername = function(inputType, displayWarning, nthChild) {
+	let submitButtonEnabled = false;
+	var form = document.forms["signupform"][inputType].value;
+	if(form.length < 5) {
+		//smg.submitButtonEnabled = false;
+		isItTrue.someval1(submitButtonEnabled);
+		valildateAll(displayWarning, nthChild);
+	}
+	else {
+		submitButtonEnabled = true;
+		//isItTrue.someval1.call(smg);
+		isItTrue.someval1(submitButtonEnabled);
+		document.querySelector("#signupform > button").disabled = false;
+		document.querySelector("#signupform span:nth-child(" + nthChild + ")").innerHTML = '';
```

Comment:
Don't use nth-child unless actually necessary. This can be better handled using id selectors.